core/src/main/scala/quasar/datasource/kafka/SeekConsumer.scala

```diff
@@ -79,54 +90,77 @@ class FullConsumer[F[_]: ConcurrentEffect: ContextShift: Timer, K, V](
   def partitionInfoToTopicPartition(info: PartitionInfo): TopicPartition =
     new TopicPartition(info.topic(), info.partition())
 
+  type CCR = CommittableConsumerRecord[F, K, V]
+
   /**
    * Emits all messages up to the last message of each partition given the map of end offsets.
    *
    * @param endOffsets Map of all _non-empty_ topic/partitions to their end offsets (offset of last message + 1)
    */
   def takeUntilEndOffsets(
-      // TODO: it would make more sense to take a KafkaConsumer, but that can't be mocked on fs2-kafka 1.0.0.
-      stream: Stream[F, Stream[F, CommittableConsumerRecord[F, K, V]]],
+      stream: Stream[F, Stream[F, CCR]],
       endOffsets: Map[TopicPartition, Long])
-      : Stream[F, CommittableConsumerRecord[F, K, V]] = {
+      : Stream[F, CCR] =
     stream
-      .take(endOffsets.size.toLong) // assumes no automatic assignment; needs a filter otherwise
-      .map(_.takeThrough(isNotOffsetLimit(_, endOffsets)))
+      .take(endOffsets.size.toLong)
+      .map(_.through(offsetPipe(endOffsets)))
```

Comment:
If we had access to the start position of each partition (i.e. where we seeked to), could we
```scala
_.takeThrough(isNotOffsetLimit(_, endOffsets)).filter(_.offset > seekPosition)
```

Do we know what happens if we attempt to seek to an offset that is less than the available range? Will it error? Postion at the beginning of the partition?

Reply:
No, we don't have `seekPosition` here unfortunately, since we don't have `partition` and can't access `initialOffsets`. If `seek` argument is negative it throws an exception.

Reply:
Indeed, we don't have access currently, but we could adjust the signature and pass in the `initialOffsets` to use in the `.filter` which I think would allow us to avoid `offsetPull`.

Reply:
Thanks! I've just realized that there won't be any halts, since we take one element more, and `.filter` will work :tada: